float64[] pose
---
float64[] q
